client/upgrade/cli/query.go

```diff
@@ -90,29 +97,46 @@ func GetCmdQuerySignals(storeName string, cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			var validatorAddrs []string
+			validatorConsAddrs := make(map[string]bool)
 			res, err := cliCtx.QuerySubspace(upgrade.GetSignalPrefixKey(upgradeConfig.Protocol.Version), storeName)
 			if err != nil {
 				return err
 			}
 
 			for _, kv := range res {
-				validatorAddrs = append(validatorAddrs, upgrade.GetAddressFromSignalKey(kv.Key))
+				validatorConsAddrs[upgrade.GetAddressFromSignalKey(kv.Key)] = true
 			}
 
-			if len(validatorAddrs) == 0 {
+			if len(validatorConsAddrs) == 0 {
 				fmt.Println("No validators have started the new version.")
 				return nil
 			}
 
-			output, err := codec.MarshalJSONIndent(cdc, validatorAddrs)
+			key := stake.ValidatorsKey
+			resKVs, err := cliCtx.QuerySubspace(key, "stake")
 			if err != nil {
 				return err
 			}
 
-			fmt.Println(string(output))
+			isDetail := viper.GetBool(flagDetail)
+			totalVotingPower := sdk.ZeroDec()
+			signalsVotingPower := sdk.ZeroDec()
+
+			for _, kv := range resKVs {
+				addr := kv.Key[1:]
+				validator := types.MustUnmarshalValidator(cdc, addr, kv.Value)
+				totalVotingPower = totalVotingPower.Add(validator.GetPower())
+				if _, ok := validatorConsAddrs[validator.GetConsAddr().String()]; ok {
+					signalsVotingPower = signalsVotingPower.Add(validator.GetPower())
+					if isDetail {
+						fmt.Println(validator.GetOperator().String(), " ", validator.GetPower().String())
+					}
+				}
+			}
+			fmt.Println("siganalsVotingPower/totalVotingPower = " + signalsVotingPower.Quo(totalVotingPower).String())
 			return nil
 		},
 	}
+	cmd.Flags().Bool(flagDetail, false, "details of siganls")
```

Comment:
Why default value is false?

Reply:
By default, only print the siganal VotingPower percent.